src/Courses/CourseDetailsTabs/TopicsTab.tsx

```diff
@@ -417,13 +426,49 @@ export const TopicsTab: React.FC<TopicsTabProps> = ({ course, setCourse }) => {
                             {/* <span style={style} onClick={onClick} role="button" tabIndex={0} onKeyPress={onClick} > */}
                             {
                                 inEditMode &&
-                                <Button variant='outline-success'
-                                    tabIndex={0}
-                                    onClick={_.partial(addUnitClick, _, course.id)}
-                                    onKeyPress={_.partial(addUnitClick, _, course.id)}
-                                >
-                                    <FaPlusCircle /> New Unit
-                                </Button>
+                                <>
+                                    <CourseTarballImportButton 
+                                        style={{
+                                            marginLeft: '1em'
+                                        }}
+                                        courseId={course.id}
+                                        /* Can't deconstruct here because the type changes based on the status object (even though it has the same props) */
+                                        onEvent={(event) => {
+                                            // Grabbing this for error handling (see default below)
+                                            const { status } = event;
+                                            if (status !== 'error') {
+                                                setError(null);
+                                            }
+                                            if (status !== 'loading') {
+                                                setLoading(false);
+                                            }
+                                            switch (event.status) {
+                                            case 'error':
+                                                setError(event.data);
+                                                break;
+                                            case 'success':
+                                                setUnitInCourse(event.data);
+                                                break;
+                                            case 'loading':
+                                                setLoading(true);
+                                                break;
```

Comment:
This is probably a good example of where one state is better than multiple, and a reducer might help.

As it is, the `switch` doesn't add much value when it'd be more clearly expressed as `if/else`s.